File: gaia-sdk-javascript/src/graphql/request/type/Introspection.ts

import {SkillBuildJob} from "./SkillBuildJob";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

export class Introspection extends Array<(_:VariableRegistry) => string> {
public _typeName = "Introspection";
    /**
     * Introspects the build jobs currently available in the system
     */
    public buildJobs = (tenantId: Uuid, config: (_:SkillBuildJob) => void) => this.push((registry) => {
        const name1 = registry.register("tenantId", tenantId);
        const entity = new SkillBuildJob();
        config(entity);
        return `buildJobs(tenantId:${name1}){` + entity.render(registry) + "}"
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/Introspection.py

from typing import Callable

from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.type.SkillBuildJob import SkillBuildJob


class Introspection(list):

    """
    Introspects the build jobs currently available in the system
    """
    def build_jobs(self, tenantId: str, config: Callable[['SkillBuildJob'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("tenantId", tenantId)
            entity = SkillBuildJob()
            config(entity)
            return f'buildJobs(tenantId:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
